src/Stratis.Bitcoin.Features.BlockStore/Controllers/BlockStoreController.cs

```diff
@@ -289,6 +296,64 @@ public IActionResult GetUtxoSet(int atBlockHeight)
             }
         }
 
+        /// <summary>Provides verbose balance data of the given addresses.</summary>
+        /// <param name="address">The address to query for unspent outputs.</param>
+        /// <returns>A result object containing the UTXOs.</returns>
+        /// <response code="200">Returns the UTXO set for a particular address at the current chain tip height.</response>
+        /// <response code="400">Unexpected exception occurred</response>
+        [Route(BlockStoreRouteEndPoint.GetUtxoSetForAddress)]
+        [HttpGet]
+        [ProducesResponseType((int)HttpStatusCode.OK)]
+        [ProducesResponseType((int)HttpStatusCode.BadRequest)]
+        public IActionResult GetUtxoSetForAddress(string address)
+        {
+            // Get coinview at current height (SLOW)
+            var coinView = this.utxoIndexer.GetCoinviewAtHeight(this.chainIndexer.Height);
+
+            var utxos = new List<UtxoModel>();
+
+            // Get utxos for this address
+            try
+            {
+                // Fine to do this in parallel because we don't care about the order
+                Parallel.ForEach(coinView.UnspentOutputs, (utxo) =>
+                {
+                    var tx = coinView.Transactions[utxo.Hash];
+
+                    // Get the actual output
+                    var output = tx.Outputs[utxo.N];
+
+                    var destinationAddress = this.scriptAddressReader.GetAddressFromScriptPubKey(this.network, output.ScriptPubKey);
+
+                    if (destinationAddress != address)
+                        return;
+
+                    var utxoModel = new UtxoModel()
+                    {
+                        TxId = utxo.Hash,
+                        Index = utxo.N,
+                        ScriptPubKey = output.ScriptPubKey,
+                        Value = output.Value
+                    };
+
+                    utxos.Add(utxoModel);
```

Comment:
I think we have to be careful about thread safety here - if the order does indeed not matter then using one of the concurrent collections instead would be preferable

Reply:
Good call, changed to use a concurrent bag.